packages/server-core/src/projects/project/project.class.ts

```diff
@@ -133,6 +143,188 @@ const getGitProjectData = (project) => {
   return response
 }
 
+export const updateProject = async (
+  app: Application,
+  data: {
+    sourceURL: string
+    destinationURL: string
+    name?: string
+    needsRebuild?: boolean
+    reset?: boolean
+    commitSHA?: string
+    sourceBranch: string
+    updateType: ProjectUpdateType
+    updateSchedule: string
+  },
+  params?: ProjectUpdateParams
+) => {
+  if (data.sourceURL === 'default-project') {
+    copyDefaultProject()
+    await uploadLocalProjectToProvider(app, 'default-project')
+    return
+  }
+
+  const urlParts = data.sourceURL.split('/')
+  let projectName = data.name || urlParts.pop()
+  if (!projectName) throw new Error('Git repo must be plain URL')
+  projectName = projectName.toLowerCase()
+  if (projectName.substring(projectName.length - 4) === '.git') projectName = projectName.slice(0, -4)
+  if (projectName.substring(projectName.length - 1) === '/') projectName = projectName.slice(0, -1)
+
+  const projectLocalDirectory = path.resolve(appRootPath.path, `packages/projects/projects/`)
+  const projectDirectory = path.resolve(appRootPath.path, `packages/projects/projects/${projectName}/`)
+
+  // if project exists already, remove it and re-clone it
+  if (fs.existsSync(projectDirectory)) {
+    // if (isDev) throw new Error('Cannot create project - already exists')
+    deleteFolderRecursive(projectDirectory)
+  }
+
+  const project = await app.service('project').Model.findOne({
+    where: {
+      name: projectName
+    }
+  })
+
+  const userId = params!.user?.id || project.updateUserId
+
+  const githubIdentityProvider = await app.service('identity-provider').Model.findOne({
```

Comment:
Instead of using model, we can use api call here.

Reply:
Resolved